### PR TITLE
[SYCL][Doc]Editorial changes "if_architecture_is"

### PR DESCRIPTION
**Pull request by gmlueck**
This commit makes some editorial changes to the "if_architecture_is" extension specification.  None of the APIs changed, so this has no effect on the implementation.  I'm making this in preparation for some new APIs that I will soon add to this extension.

* Eliminate redundancy with the `architecture` enumeration.  The code synopsis no longer duplicates the enumerators.  These are now entirely defined in the table.

* Add some missing entries to the table defining the enumerators, and fix some obvious cut-and-paste errors with the descriptions.

* Update the function descriptions to our new format, using subsections `Constraints`, `Preconditions`, etc.

* Use our new style where we make a line break after every sentence.

### CODE REVIEW

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc [hunk 1]
```diff
@@ -94,514 +101,761 @@ implementation supports.
...
+intel_gpu_9_6_0 = intel_gpu_aml
+intel_gpu_9_7_0 = intel_gpu_cml
+intel_gpu_11_0_0 = intel_gpu_icllp
+intel_gpu_12_0_0 = intel_gpu_tgllp
+intel_gpu_12_10_0 = intel_gpu_dg1
+----
+|-
+|Aliases for Intel graphics architectures.
+
+3+^|*Nvidia GPU family*
```
Comment (Pennycook) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:353]:
```suggestion
3+^|*NVIDIA GPU family*
```
...to match the rest of the table.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc [hunk 2]
```diff
@@ -94,514 +101,761 @@ implementation supports.
...
 } // namespace sycl::ext::oneapi::experimental
-```
+----
+!====
 
-This function operates exactly like `if_device_has` from the
+_Constraints:_ The type `T` must be a {cpp} `Callable` type which is invocable
+with an empty parameter list.
+
+_Preconditions:_ This function must not be called from host code.
```
Comment (Pennycook) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:655]: Nit: I don't know if we have a precedent here, but I usually don't write these as negated conditions.

```suggestion
_Preconditions:_ This function must be called from device code.
```

**Review by Pennycook — APPROVED** [review]
Two small and optional suggestions.

**Comment by gmlueck** [comment]
@intel/llvm-gatekeepers I think this PR is ready to merge.

**Pull request merged**